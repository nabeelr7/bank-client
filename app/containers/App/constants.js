/*
 *
 * LoginPage constants
 *
 */

export const LOGOUT_REQUEST = 'app/App/LOGOUT_REQUEST';
export const LOGOUT_SUCCESS = 'app/App/LOGOUT_SUCCESS';
export const LOGOUT_ERROR = 'app/App/LOGOUT_ERROR';

export const COLLAPSED_SIDEBAR = 'app/App/COLLAPSED_SIDEBAR';
export const COLLAPSED_DRAWER = 'app/App/COLLAPSED_DRAWER';

export const GET_CURRENCIES_REQUEST = 'app/App/GET_CURRENCIES_REQUEST';
export const GET_CURRENCIES_SUCCESS = 'app/App/GET_CURRENCIES_SUCCESS';
export const GET_CURRENCIES_ERROR = 'app/SettingsPage/GET_CURRENCIES_ERROR';

export const SELECT_CURRENCY = 'app/App/SELECT_CURRENCY';

export const NEXT_STEP = 'app/App/NEXT_STEP';
export const PREVIOUS_STEP = 'app/App/PREVIOUS_STEP';

export const CHANGE_INPUT_NUMBER = 'app/App/CHANGE_INPUT_NUMBER';
export const CHANGE_INPUT = 'app/App/CHANGE_INPUT';

export const TOGGLE_MODAL = 'app/DashboardPage/TOGGLE_MODAL';
