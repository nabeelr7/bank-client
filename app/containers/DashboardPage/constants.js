/*
 *
 * DashboardPage constants
 *
 */

export const GET_CURRENCIES_REQUEST =
  'app/DashboardPage/GET_CURRENCIES_REQUEST';
export const GET_CURRENCIES_SUCCESS =
  'app/DashboardPage/GET_CURRENCIES_SUCCESS';
export const GET_CURRENCIES_ERROR = 'app/DashboardPage/GET_CURRENCIES_ERROR';

export const SELECT_CURRENCY = 'app/DashboardPage/SELECT_CURRENCY';

export const CREATE_NEW_BILL_REQUEST =
  'app/DashboardPage/CREATE_NEW_BILL_REQUEST';
export const CREATE_NEW_BILL_SUCCESS =
  'app/DashboardPage/CREATE_NEW_BILL_SUCCESS';
export const CREATE_NEW_BILL_INCORRECT =
  'app/DashboardPage/CREATE_NEW_BILL_INCORRECT';

export const GET_BILLS_REQUEST = 'app/DashboardPage/GET_BILLS_REQUEST';
export const GET_BILLS_SUCCESS = 'app/DashboardPage/GET_BILLS_SUCCESS';
export const GET_BILLS_ERROR = 'app/DashboardPage/GET_BILLS_ERROR';

export const GET_ACCOUNT_BALANCE_REQUEST =
  'app/DashboardPage/GET_ACCOUNT_BALANCE_REQUEST';
export const GET_ACCOUNT_BALANCE_SUCCESS =
  'app/DashboardPage/GET_ACCOUNT_BALANCE_SUCCESS';
export const GET_ACCOUNT_BALANCE_ERROR =
  'app/DashboardPage/GET_ACCOUNT_BALANCE_ERROR';

export const GET_RECENT_TRANSACTIONS_REQUEST =
  'app/DashboardPage/GET_RECENT_TRANSACTIONS_REQUEST';
export const GET_RECENT_TRANSACTIONS_SUCCESS =
  'app/DashboardPage/GET_RECENT_TRANSACTIONS_SUCCESS';
export const GET_RECENT_TRANSACTIONS_ERROR =
  'app/DashboardPage/GET_RECENT_TRANSACTIONS_ERROR';

export const CHANGE_LAYOUT = 'app/DashboardPage/CHANGE_LAYOUT';

export const TOGGLE_MODAL = 'app/DashboardPage/TOGGLE_MODAL';

export const GET_AVAILABLE_FUNDS_REQUEST =
  'app/DashboardPage/GET_AVAILABLE_FUNDS_REQUEST';
export const GET_AVAILABLE_FUNDS_SUCCESS =
  'app/DashboardPage/GET_AVAILABLE_FUNDS_SUCCESS';
export const GET_AVAILABLE_FUNDS_ERROR =
  'app/DashboardPage/GET_AVAILABLE_FUNDS_ERROR';
