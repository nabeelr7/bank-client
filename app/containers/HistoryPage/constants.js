/*
 *
 * HistoryPage constants
 *
 */

export const GET_TRANSFER_HISTORY_REQUEST =
  'app/HistoryPage/GET_TRANSFER_HISTORY_REQUEST';
export const GET_TRANSFER_HISTORY_SUCCESS =
  'app/HistoryPage/GET_TRANSFER_HISTORY_SUCCESS';
export const GET_TRANSFER_HISTORY_ERROR =
  'app/HistoryPage/GET_TRANSFER_HISTORY_ERROR';
