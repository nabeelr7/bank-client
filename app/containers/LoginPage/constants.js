/*
 *
 * LoginPage constants
 *
 */

export const CHANGE_PINCODE = 'app/LoginPage/CHANGE_PINCODE';
export const CHANGE_PASSWORD = 'app/LoginPage/CHANGE_PASSWORD';

export const NEXT_STEP = 'app/LoginPage/NEXT_STEP';
export const PREVIOUS_STEP = 'app/LoginPage/PREVIOUS_STEP';

export const LOGIN = 'app/LoginPage/LOGIN';
export const LOGIN_SUCCESS = 'app/LoginPage/LOGIN_SUCCESS';
export const LOGIN_ERROR = 'app/LoginPage/LOGIN_ERROR';
