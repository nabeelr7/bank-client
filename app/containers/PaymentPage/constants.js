/*
 *
 * PaymentPage constants
 *
 */

export const GET_BILLS_REQUEST = 'app/PaymentPage/GET_BILLS_REQUEST';
export const GET_BILLS_SUCCESS = 'app/PaymentPage/GET_BILLS_SUCCESS';
export const GET_BILLS_ERROR = 'app/PaymentPage/GET_BILLS_ERROR';

export const SEARCH_RECIPIENT_REQUEST =
  'app/PaymentPage/SEARCH_RECIPIENT_REQUEST';
export const SEARCH_RECIPIENT_SUCCESS =
  'app/PaymentPage/SEARCH_RECIPIENT_SUCCESS';
export const SEARCH_RECIPIENT_ERROR = 'app/PaymentPage/SEARCH_RECIPIENT_ERROR';

export const CHECK_RECIPIENT = 'app/PaymentPage/CHECK_RECIPIENT';
export const CHECK_RECIPIENT_CORRECT =
  'app/PaymentPage/CHECK_RECIPIENT_CORRECT';
export const CHECK_RECIPIENT_INCORRECT =
  'app/PaymentPage/CHECK_RECIPIENT_INCORRECT';

export const CREATE_TRANSACTION_REQUEST =
  'app/PaymentPage/CREATE_TRANSACTION_REQUEST';
export const CREATE_TRANSACTION_SUCCESS =
  'app/PaymentPage/CREATE_TRANSACTION_SUCCESS';
export const CREATE_TRANSACTION_ERROR =
  'app/PaymentPage/CREATE_TRANSACTION_ERROR';

export const GET_AUTHORIZATION_KEY_REQUEST =
  'app/PaymentPage/GET_AUTHORIZATION_KEY_REQUEST';
export const GET_AUTHORIZATION_KEY_SUCCESS =
  'app/PaymentPage/GET_AUTHORIZATION_KEY_SUCCESS';
export const GET_AUTHORIZATION_KEY_ERROR =
  'app/PaymentPage/GET_AUTHORIZATION_KEY_ERROR';

export const CONFIRM_TRANSACTION_REQUEST =
  'app/PaymentPage/CONFIRM_TRANSACTION_REQUEST';
export const CONFIRM_TRANSACTION_SUCCESS =
  'app/PaymentPage/CONFIRM_TRANSACTION_SUCCESS';
export const CONFIRM_TRANSACTION_ERROR =
  'app/PaymentPage/CONFIRM_TRANSACTION_ERROR';

export const CHANGE_INPUT = 'app/PaymentPage/CHANGE_INPUT';
export const CHANGE_INPUT_NUMBER = 'app/PaymentPage/CHANGE_INPUT_NUMBER';
export const SELECT_SENDER_BILL = 'app/PaymentPage/SELECT_SENDER_BILL';

export const NEXT_STEP = 'app/PaymentPage/NEXT_STEP';
export const PREVIOUS_STEP = 'app/PaymentPage/PREVIOUS_STEP';
