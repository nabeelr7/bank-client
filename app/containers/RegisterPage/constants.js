/*
 *
 * RegisterPage constants
 *
 */

export const GET_CURRENCIES = 'app/RegisterPage/GET_CURRENCIES';
export const GET_CURRENCIES_SUCCESS = 'app/RegisterPage/GET_CURRENCIES_SUCCESS';
export const GET_CURRENCIES_ERROR = 'app/RegisterPage/GET_CURRENCIES_ERROR';
