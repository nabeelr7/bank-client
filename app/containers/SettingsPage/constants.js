/*
 *
 * SettingsPage constants
 *
 */

export const GET_USER_DATA_REQUEST = 'app/SettingsPage/GET_USER_DATA_REQUEST';
export const GET_USER_DATA_SUCCESS = 'app/SettingsPage/GET_USER_DATA_SUCCESS';
export const GET_USER_DATA_ERROR = 'app/SettingsPage/GET_USER_DATA_ERROR';
