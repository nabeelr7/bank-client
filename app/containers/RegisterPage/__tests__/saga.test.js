import { takeLatest, put } from 'redux-saga/effects';
import registerPageSaga, { getCurrencies, register, checkEmail } from '../saga';
import { GET_CURRENCIES, REGISTER, CHECK_EMAIL } from '../constants';
import {
  getCurrenciesSuccessAction,
  getCurrenciesErrorAction,
} from '../actions';

describe('getCurrencies Saga', () => {
  let getCurrenciesGenerator;

  beforeEach(() => {
    getCurrenciesGenerator = getCurrencies();

    const selectDescriptor = getCurrenciesGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the getCurrenciesSuccess action if it requests the data successfully', () => {
    const response = {
      data: [
        {
          uuid: 'f464c93d-d9cd-4240-9d39-f524d559df83',
          name: 'EUR',
          currentExchangeRate: 0.218981299,
        },
        {
          uuid: '0880acd7-f04b-49d6-8022-ec9b61a722e8',
          name: 'USD',
          currentExchangeRate: 0.2363246179,
        },
        {
          uuid: '0d3c9ec3-4cad-4952-b3d1-53da66faefa5',
          name: 'PLN',
          currentExchangeRate: 1,
        },
      ],
    };
    const putDescriptor = getCurrenciesGenerator.next(response).value;
    expect(putDescriptor).toEqual(
      put(getCurrenciesSuccessAction(response.data)),
    );
  });

  it('should call the getCurrenciesError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getCurrenciesGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(getCurrenciesErrorAction(response)));
  });
});

const value = 'adrian@test.com';
const reject = {};
const resolve = {};

describe('checkEmail Saga', () => {
  let checkEmailGenerator;

  beforeEach(() => {
    checkEmailGenerator = checkEmail();

    const selectDescriptor = checkEmailGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = checkEmailGenerator.next({ value, reject, resolve })
      .value;
    expect(callDescriptor).toMatchSnapshot();
  });
});

describe('register Saga', () => {
  let registerGenerator;

  beforeEach(() => {
    registerGenerator = register();

    const selectDescriptor = registerGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = registerGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });
});

describe('registerPageSaga Saga', () => {
  const registerPage = registerPageSaga();

  it('should start task to watch for GET_CURRENCIES action', () => {
    const takeLatestDescriptor = registerPage.next().value;
    expect(takeLatestDescriptor).toEqual(
      takeLatest(GET_CURRENCIES, getCurrencies),
    );
  });

  it('should start task to watch for REGISTER action', () => {
    const takeLatestDescriptor = registerPage.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(REGISTER, register));
  });

  it('should start task to watch for CHECK_EMAIL action', () => {
    const takeLatestDescriptor = registerPage.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(CHECK_EMAIL, checkEmail));
  });
});
