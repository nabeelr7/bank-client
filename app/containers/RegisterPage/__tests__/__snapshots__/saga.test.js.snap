// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCurrencies Saga should dispatch the getCurrenciesSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:4000/api/Currencies",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
